File: lambda/lambda.js
// findSum accepts two variables and a function
// combining the two variables and passing that 
// into the passed function
function findSum(x, y, cb) {
    var result = x + y;
    cb(result);
}

// here we call findSum passing two numbers and a 
// lambda function that creates an alert box showing
// the sum
findSum(1, 2, function(x) {
    window.alert(x);
});

File: lambda/lambda.hs
-- findSum accepts two variables and a function
-- combining the two variables and passing that 
-- into the passed function
findSum :: Integer -> Integer -> (Integer -> Integer) -> Integer
findSum x y f = (f(x+y)) 

main = 
    -- here we are calling findSum and passing two variables
    -- and an anonymous function that accepts one variable
    -- then doubles it
    print(findSum 2 3 (\x -> x * 2))
